sqlstore/src/main/java/uk/ac/bbsrc/tgac/miso/persistence/impl/HibernateSampleNumberPerProjectDao.java

```diff
@@ -63,4 +67,45 @@ public void update(SampleNumberPerProject sampleNumberPerProject) {
     currentSession().update(sampleNumberPerProject);
   }
 
+  @Override
+  public synchronized String nextNumber(Project project, User user) {
+    Query query = currentSession().createQuery("from SampleNumberPerProjectImpl sn where sn.project = :project");
+    query.setParameter("project", project);
+    SampleNumberPerProject sampleNumberPerProject = (SampleNumberPerProject) query.uniqueResult();
+    if (sampleNumberPerProject == null) {
+      sampleNumberPerProject = createSampleNumberPerProject(project, user);
+    }
+    Integer highestSampleNumber = sampleNumberPerProject.getHighestSampleNumber();
+    highestSampleNumber++;
+    sampleNumberPerProject.setHighestSampleNumber(highestSampleNumber);
+    sampleNumberPerProject.setUpdatedBy(user);
+    sampleNumberPerProject.setLastUpdated(new Date());
+    addSampleNumberPerProject(sampleNumberPerProject);
+
+    return padInteger(sampleNumberPerProject.getPadding(), highestSampleNumber);
+  }
+
+  private SampleNumberPerProject createSampleNumberPerProject(Project project, User user) {
+    SampleNumberPerProject sampleNumberPerProject;
+    sampleNumberPerProject = new SampleNumberPerProjectImpl();
+    sampleNumberPerProject.setProject(project);
+    sampleNumberPerProject.setHighestSampleNumber(0);
+    sampleNumberPerProject.setPadding(DEFAULT_PADDING);
+    sampleNumberPerProject.setCreatedBy(user);
+    sampleNumberPerProject.setUpdatedBy(user);
+    sampleNumberPerProject.setLastUpdated(new Date());
+    sampleNumberPerProject.setCreationDate(new Date());
```

Comment:
Setting the dates is handled inside the addSampleNumberPerProject method, so is not required here.